tac/agents/v1/base/reactions.py

```diff
@@ -398,10 +401,11 @@ def on_unidentified_dialogue(self, msg: AgentMessage) -> None:
         :return: None
         """
         logger.debug("[{}]: Unidentified dialogue.".format(self.agent_name))
-        result = ByteMessage(msg.msg_id + 1, msg.dialogue_id, msg.destination, b'This message belongs to an unidentified dialogue.', Context())
-        self.out_box.out_queue.put(result)
+        result = ByteMessage(to=msg.sender, sender=self.crypto.public_key, message_id=msg.get("id") + 1,
```

Comment:
this will break if msg.get("id") does not return an integer. Given that you're dealing with an unidentified dialogue that's likely!